lib/spack/spack/util/azure_blob.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""This file contains the definition of the Azure Blob storage Class used to
+   integrate Azure Blob storage with spack buildcache.
+"""
+
+import datetime
+import os
+
+import llnl.util.tty as tty
+
+
+class AzureBlob:
+
+    def __init__(self, url):
+        from azure.storage.blob import BlobServiceClient
+        self.url = url
+        (self.container_name, self.blob_path) = self.get_container_blob_path()
+        if url.scheme != 'azure':
+            raise ValueError(
+                'Can not create Azure blob connection from URL with scheme: %s'
+                % (url.scheme))
+        if "AZURE_STORAGE_CONNECTION_STRING" in os.environ:
+            self.connect_str = (os.environ.
+                                get('AZURE_STORAGE_CONNECTION_STRING'))
+            self.blob_service_client = (BlobServiceClient.
+                                        from_connection_string(self.
+                                                               connect_str))
+            if not self.azure_container_exists():
+                tty.warn("The container %s does not exist, it will be created"
+                         % (self.container_name))
+                self.blob_service_client.create_container(self.container_name)
+        else:
+            tty.error("Error: Environmental variable \
+ AZURE_STORAGE_CONNECTION_STRING is not defined, it is \
+ required if you want to use Azure Blob storage as an \
+ spack buildcache")
+
+    def get_container_blob_path(self):
+        blob_path = self.url.path
+        path_list = blob_path.split('/')
+        container_name = path_list[1]
+        blob_path = os.path.join(*path_list[2:])
+        tty.debug("container_name = %s, blob_path = %s"
+                  % (container_name, blob_path))
+        return (container_name, blob_path)
+
+    def is_https(self):
+        if 'https' in self.connect_str:
+            return True
+        else:
+            return False
```

Comment:
Would this one work:
```suggestion
        return self.connect_str.startswith('https')
```
?

Reply:
'https' is not at the beginning of the connection string. (startswith will not work in this case)